### PR TITLE
config: add missing nodelay example

### LINKED ISSUE #35254: There is no corresponding example for tcp nodelay option
## Enhancement

Option introduced by #22757 does not have corresponding field in `config.toml.example`

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: close #35254

Problem Summary: Add missing example

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: config/config.toml.example [hunk 1]
```diff
@@ -226,6 +226,9 @@ stmt-count-limit = 5000
 # Set keep alive option for tcp connection.
 tcp-keep-alive = true
 
+# Set nodelay option for tcp connection.
+tcp-no-delay = false
```
Comment (ran-huang) [inline config/config.toml.example:230]: This looks inconsistent with the user documentation: https://docs.pingcap.com/tidb/dev/tidb-configuration-file#tcp-no-delay
Can you confirm which default value is correct?

**Review by ran-huang — APPROVED** [review]
LGTM. Thanks!

**Comment by hawkingrei** [comment]
@bb7133 @XuHuaiyu PTAL

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**